### PR TITLE
Task inputs/outputs runtime API documentation should always configure property names and normalization

### LINKED ISSUE #7978: Task inputs/outputs runtime API documentation should always configure property names and normalization
https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:task_input_output_runtime_api

We have examples of just using `inputs.files()` with no normalization or property name. We should change this and recommend to always specify these. The justification should be that this provides better diagnostics via the inputs logging and inputs comparison feature of Gradle Enterprise.

### PR DESCRIPTION
**Pull request by paplorinc**
Fixes #7978

### CODE REVIEW

File: subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle [hunk 1]
```diff
@@ -20,8 +20,12 @@ repositories {
 task createClasspathManifest {
     def outputDir = file("$buildDir/$name")
 
-    inputs.files sourceSets.main.runtimeClasspath
-    outputs.dir outputDir
+    inputs.files(sourceSets.main.runtimeClasspath)
+        .withPropertyName("runtimeClasspath")
+        .withPathSensitivity(PathSensitivity.RELATIVE)
+        .withNormalizer(ClasspathNormalizer)
```
Comment (paplorinc) [inline subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle]: @lptr, does it make sense to add a classpath normalizer here?
I didn't see any such example in our codebase

File: subprojects/docs/src/samples/userguide/tasks/incrementalBuild/customTaskClass/groovy/build.gradle [hunk 2]
```diff
@@ -1,5 +1,3 @@
-import org.example.*
```
Comment (paplorinc) [inline subprojects/docs/src/samples/userguide/tasks/incrementalBuild/customTaskClass/groovy/build.gradle]: was this used?

File: subprojects/internal-performance-testing/internal-performance-testing.gradle.kts [hunk 3]
```diff
@@ -86,6 +86,9 @@ java.sourceSets.main { output.dir(mapOf("builtBy" to reportResources), generated
 
 tasks.jar {
     inputs.files(flamegraph)
+        .withPropertyName("flamegraph")
+        .withPathSensitivity(PathSensitivity.RELATIVE)
```
Comment (paplorinc) [inline subprojects/internal-performance-testing/internal-performance-testing.gradle.kts]: @oehme, do you think this makes sense?

File: subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle (hunk 1 above)
Comment (lptr) [reply subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle]: It does, and it makes the `.withPathSensitivity(PathSensitivity.RELATIVE)` redundant.

Comment (lptr) [reply subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle]: Did you check the documentation text from where we link to this?

Comment (paplorinc) [reply subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle]: Checked the javadocs that didn't answer the above question.
Do you mean any specific doc I could check nex time?
Removed `withPathSensitivity` from here

File: subprojects/internal-performance-testing/internal-performance-testing.gradle.kts (hunk 3 above)
Comment (oehme) [reply subprojects/internal-performance-testing/internal-performance-testing.gradle.kts]: yep

File: subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle (hunk 1 above)
Comment (lptr) [reply subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle]: Sorry, what I meant was: could you check the doc that links to this sample, and see if it needs adjusting?

To answer your question all should be documented (albeit badly) here: https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:task_input_output_annotations

**Comment by paplorinc** [comment]
Rebased to simplify keeping track of some reverts and stuff.

File: subprojects/docs/src/samples/userguide/tasks/incrementalBuild/customTaskClass/groovy/build.gradle (hunk 2 above)
Comment (paplorinc) [reply subprojects/docs/src/samples/userguide/tasks/incrementalBuild/customTaskClass/groovy/build.gradle]: Yes, it was, reverted

File: subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle (hunk 1 above)
Comment (lptr) [reply subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle]: Did you get to look at `test_kit.adoc`?

Comment (lptr) [reply subprojects/docs/src/samples/testKit/gradleRunner/manualClasspathInjection/groovy/build.gradle]: We talked about this and the doc has been checked.

**Comment by paplorinc** [comment]
Cherry-picked to `release`, too.

**Pull request merged**